pandas/core/groupby/ops.py

```diff
@@ -634,10 +687,16 @@ def _cython_operation(
         # if not raise NotImplementedError
         cy_op.disallow_invalid_ops(dtype, is_numeric)
 
+        func_uses_mask = cy_op.uses_mask()
         if is_extension_array_dtype(dtype):
-            return self._ea_wrap_cython_operation(
-                kind, values, how, axis, min_count, **kwargs
-            )
+            if isinstance(values, BaseMaskedArray) and func_uses_mask:
```

Comment:
can we do this check at the top of _ea_wrap_cython_operation instead of here?

Reply:
Currently, `_ea_wrap_cython_operation` doesn't have access to `cy_op`, so that or `func_uses_mask` would need to be passed then. 
(personally I think it's fine to keep it here)